frontend_tests/node_tests/recent_topics.js

```diff
@@ -0,0 +1,208 @@
+var rt = zrequire('recent_topics');
+
+var page_params = {
+    user_id: 1,
+};
+set_global('page_params', page_params);
+
+(function test_process_message_for_senders() {
+    var stream1 = 1;
+
+    var topic1 = "topic-1";
+    var topic2 = "topic-2";
+    var topic3 = "topic-3";
+    var topic4 = "topic-4";
+    var topic5 = "topic-5";
+    var topic6 = "topic-6";
+
+    var sender1 = 1;
+    var sender2 = 2;
+
+    // New stream
+    var messages = [];
+    messages[0] = {
+        stream_id: stream1,
+        timestamp: _.uniqueId(),
+        subject: topic1,
+        sender_id: sender1,
+        flags: ['read'],
+        type: 'stream',
+    };
+
+    messages[1] = {
+        stream_id: stream1,
+        timestamp: _.uniqueId(),
+        subject: topic2,
+        sender_id: sender1,
+        flags: ['read'],
+        type: 'stream',
+    };
+    messages[2] = {
+        stream_id: stream1,
+        timestamp: messages[1].timestamp + 1,
+        subject: topic2,
+        sender_id: sender2,
+        flags: ['read'],
+        type: 'stream',
+    };
+
+    messages[3] = {
+        stream_id: stream1,
+        timestamp: _.uniqueId(),
+        subject: topic3,
+        sender_id: sender2,
+        flags: [],
+        type: 'stream',
+    };
+
+    messages[4] = {
+        stream_id: stream1,
+        timestamp: _.uniqueId(),
+        subject: topic4,
+        sender_id: sender1,
+        flags: ['read'],
+        type: 'stream',
+    };
+    messages[5] = {
+        stream_id: stream1,
+        timestamp: messages[4].timestamp + 1,
+        subject: topic4,
+        sender_id: sender2,
+        flags: [],
+        type: 'stream',
+    };
+
+    messages[6] = {
+        stream_id: stream1,
+        timestamp: _.uniqueId(),
```

Comment:
I'd consider just using hard coded timestamps in the test, so it's more obviously deterministic.